parquet-column/src/main/java/parquet/io/api/RecordMaterializer.java

```diff
@@ -33,6 +35,7 @@
 
   /**
    * @return the result of the conversion
+   * @throws CorruptRecordException to signal that a record is corrupt, but can be skipped
```

Comment:
Is "Corrupt" the correct term here? The data is not corrupted but no valid instance of T can be produced for the current record and this implementation of RecordMaterializer.